firebase/test/core/utils/deep-equal.spec.ts

```diff
@@ -0,0 +1,22 @@
+import * as assert from "assert";
+import { objectsEqual } from "#utils/deep-equals";
+
+describe("objectsEqual", () => {
+  it("should return true when two objects are strictly equal", () => {
+    const objectA = { a: { n: 0 } };
+    const objectB = { a: { n: 0 } };
+
+    const deepEqual = objectsEqual(objectA, objectB);
+
+    assert.ok(deepEqual);
+  });
+
+  it("should return false when two objects aren't strictly equal", () => {
+    const objectA = { a: { n: 0 } };
+    const objectB = { a: { n: "0" } };
+
+    const deepEqual = objectsEqual(objectA, objectB);
+
+    assert.ok(!deepEqual);
```

Comment:
don't we have a NotOk or something?

Reply:
Unfortunately not. The only ones that I've found are:

```typescript
            | 'equal'
            | 'notEqual'
            | 'deepEqual'
            | 'notDeepEqual'
            | 'ok'
            | 'strictEqual'
            | 'deepStrictEqual'
            | 'ifError'
            | 'strict'
```